tincr/io/design/tincr_checkpoints.tcl

```diff
@@ -178,6 +181,24 @@ proc ::tincr::read_tcp {args} {
             ::tincr::print_verbose "Done routing...($diff_time seconds)"
         }
     }
+	
+    # Complete the route for nets with a hierarchical source port.
+    # The same warning/bug described above occurs when trying to specify the ROUTE string of a net
+    # that is a hierarchical port (placed or unplaced port with no driver).
+    # Work around is also to have Vivado route these nets for us.
+    if {$link_mode=="out_of_context"} {
+	set diff_time 0
+	set hier_nets [get_nets -of [get_ports] -filter {ROUTE_STATUS == HIERPORT} -quiet]
+	    
+	if {[llength $hier_nets] > 0 } {
+	    ::tincr::print_verbose "Routing [llength $hier_nets] hierarchical port nets..."		    	    
+	    set start_time [clock microseconds]
+	    route_design -quiet -nets $hier_nets
+	    set end_time [clock microseconds]
+	    set diff_time [::tincr::format_time [expr $end_time - $start_time] s]
```

Comment:
Tincr has a method [`report_runtime`](https://github.com/byuccl/tincr/blob/master/tincr/cad/util/util.tcl#L1292) that will do all this for you:
```tcl
set diff_time [tincr::report_runtime "route_design -quiet -nets $hier_nets" s]
```

Reply:
The section of code directly above this dealt with the same problem. The comment says that ```format_time``` doesn't work for this route design command , although it should really say ```report_runtime``` doesn't work for this command.

I tried to use ```report_runtime``` for this route_design command, but also couldn't get it working. When the route_design command is either executed with ```eval``` or ```uplevel```, Vivado will encounter an error when trying to route the nets. If you are trying to only route one net, the error is something like this:
```ERROR: [Common 17-161] Invalid option value 'sw[0]' specified for 'nets'.```

If you are trying to route more than one net, the error is something like this:
``` ERROR: [Common 17-165] Too many positional options when parsing 'sw[1]', please type 'route_design -help' for usage info.``` 

This is with Vivado 2016.2.